src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ActivityLogRazorLogger.cs

```diff
@@ -51,7 +51,9 @@ public override void LogVerbose(string message)
             Log(__ACTIVITYLOG_ENTRYTYPE.ALE_INFORMATION, message);
         }
 
+#pragma warning disable VSTHRD100 // Avoid async void methods
```

Comment:
Because this is code that runs in VS, I'm going to point to the VS threading rules, specifically [Rule 4: never define async void methods](https://github.com/microsoft/vs-threading/blob/main/doc/threading_rules.md#rule-4-never-define-async-void-methods-make-the-methods-return-task-instead)

in this case, instead of doing `await _joinableTaskFactory.SwitchToMainThreadAsync();`, the code should synchronously call
```
_joinableTaskFactory.Run(async() => {
    await await _joinableTaskFactory.SwitchToMainThreadAsync();
    ...
}
```

Reply:
Or perform the work asynchronously with `_joinableTaskFactory.RunAsync(...)`, because you don't need to block for logging.